dagshub/data_engine/datasources.py

```diff
@@ -30,18 +30,19 @@ def create_datasource(repo: str, name: str, path: str, revision: Optional[str] =
 
     parsed = urllib.parse.urlparse(path)
 
-    if parsed.scheme == "" \
-        and parsed.hostname == "" \
-        and parsed.query == "" \
-        and parsed.params == "" \
-        and parsed.fragment == "":
+    if parsed.path \
+       and not parsed.scheme \
+       and not parsed.hostname \
+       and not parsed.query \
+       and not parsed.params \
+       and not parsed.fragment:
         return create_from_repo(repo, name, path=parsed.path, revision=revision)
```

Comment:
I'm not sure that this will _actually_ work, because what happens when I want to create a datasource from a repo path "www.google.com/hello"?
I'm a wacky little guy and decided to name my folders like that (I'm scraping websites maybe).
This will make us think this is a bucket URL, when in actuality it is a repo path.

I think that the way to go here should be to check against a regex that I already wrote. Sorry for not mentioning it on the last review.

https://github.com/DagsHub/client/blob/a6a757a1d0805b4551552989928498cf2355ee28/dagshub/data_engine/model/datasource_state.py#L22

Reply:
So you're saying that if it doesn't match the bucket regex, then assume it's a path?
I think your regex also won't work, e.g. if there are any dots in the bucket name?

Reply:
I checked online while I wrote this regex and while it's ok to have dots in the bucket name, it apparently breaks a lot of stuff so it's not recommended.
Thinking about it, I probably shouldn't be that opinionated about users' bucket names, so maybe it's worth changeing.

But yea, the idea is check against the regex, if it passes then it's a bucket (need to be a full match though, I see that I don't have `^` and `$` in these, which might be a mistake?)

I did write some tests for this, so you can change the regex to whatever you want as long as they pass: https://github.com/DagsHub/client/blob/a6a757a1d0805b4551552989928498cf2355ee28/tests/data_engine/test_datasource_state.py

Reply:
Or underscores